polars/polars-lazy/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -7,18 +7,22 @@ use polars_time::prelude::*;
 use super::*;
 
 #[cfg(feature = "date_offset")]
-pub(super) fn date_offset(s: Series, offset: Duration) -> PolarsResult<Series> {
+pub(super) fn date_offset(s: &[Series]) -> PolarsResult<Series> {
+    let sa = &s[0];
+    let sb = &s[1];
+
     let preserve_sortedness: bool;
-    let out = match s.dtype().clone() {
+    let out = match sa.dtype().clone() {
         DataType::Date => {
-            let s = s
+            let sa = sa
                 .cast(&DataType::Datetime(TimeUnit::Milliseconds, None))
                 .unwrap();
             preserve_sortedness = true;
-            date_offset(s, offset).and_then(|s| s.cast(&DataType::Date))
+            date_offset(&[sa, sb.clone()]).and_then(|s| s.cast(&DataType::Date))
         }
         DataType::Datetime(tu, tz) => {
-            let ca = s.datetime().unwrap();
+            let ca = sa.datetime().unwrap();
+            let offsets = sb.utf8().unwrap();
 
             fn offset_fn(tu: TimeUnit) -> fn(&Duration, i64, Option<&Tz>) -> PolarsResult<i64> {
```

Comment:
I realize this function was already there, but I don't understand why we don't assign `offset_fn` directly?

Reply:
Ah right, just this I guess:
```python
let offset_fn = match tu {
    TimeUnit::Nanoseconds => Duration::add_ns,
    TimeUnit::Microseconds => Duration::add_us,
    TimeUnit::Milliseconds => Duration::add_ms,
};
```